lib/spack/spack/test/cmd/url.py

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+# import pytest
+from spack.cmd.url import *
+
+
+# @pytest.fixture(scope='url')
+# def parser():
+#     """Returns the parser for the ``url`` command"""
+#     parser = argparse.ArgumentParser()
+#     setup_parser(parser)
+#     return parser
+
+
+class Package:
+    def __init__(self, name, versions):
+        self.name = name
+        self.versions = versions
+
+
+def test_name_parsed_correctly():
+    # Expected True
+    assert name_parsed_correctly(Package('netcdf',         []), 'netcdf')
+    assert name_parsed_correctly(Package('r-devtools',     []), 'devtools')
+    assert name_parsed_correctly(Package('py-numpy',       []), 'numpy')
+    assert name_parsed_correctly(Package('octave-splines', []), 'splines')
```

Comment:
I was trying to figure out how to use actual package classes, but I couldn't figure it out.